src/python/pants/backend/jvm/tasks/scaladoc_gen.py

```diff
@@ -59,9 +64,46 @@ def create_scaladoc_command(self, classpath, gendir, *targets):
 
     args.extend(sources)
 
+    # Check the size of the ScalaDoc command arguments, and use a options file if too long.
+    #
+    # There are a couple practical limits to consider.
+    #
+    # - The length of any individual value must be less than MAX_ARG_STRLEN
+    #   which is hardcoded to 131072 bytes.
+    #
+    # - The total length of the command must be less than the OS's `getconf ARG_MAX`
+    #   value.
+    #
+    #   This value will vary by system. For example, on my MacOS 10.13.6 Twitter-issued
+    #   developer laptop this value is currently 262144. Doing some research on the
+    #   internet shows a wide range of default values for this ranging from 131072 to
+    #   2097152 (2MB). Unfortunately, there doesn't seem to be an easy way to query the
+    #   system for this value. It's not in sysconfig.get_config_vars, or posix.environ,
+    #   or os.environ. We could shell out to `getconf` to get the actual value, but
+    #   anecdotal evidence suggests that there are other layers of limits beyond just
+    #   this POSIX value. POSIX requires ARG_MAX to be at least 4096 bytes, but it's
+    #   extremely unlikely to find a system with that low setting. Rather, I think a
+    #   practical middle ground would be the MAX_ARG_STRLEN fixed value.
+    #
+    #   So, given that rationale, if the arg list is longer than MAX_ARG_STRLEN
+    #   we should put it in an external file, just in case.
+
+    args_joined = ' '.join(map(str, args))
+    if len(args_joined) > MAX_ARG_STRLEN:
+      self.context.log.debug("ScalaDoc arguments too long. Using options file.")
+
+      # dump to file, overwriting any file that already exists
+      options_filepath = os.path.join(self.workdir, 'scaladoc_options.txt')
+      with open(options_filepath, 'w') as filehandle:
+        filehandle.write(args_joined)
+
+      args = ['@{}'.format(options_filepath)]
+
     java_executor = SubprocessExecutor(DistributionLocator.cached())
     runner = java_executor.runner(jvm_options=self.jvm_options,
                                   classpath=tool_classpath,
                                   main='scala.tools.nsc.ScalaDoc',
                                   args=args)
+
+    self.context.log.debug("SCALADOCS COMMAND: {}".format(runner.command))
```

Comment:
FTR: Fine to add this, but it should also be rendered in `./pants server --open`, and possible to extract that way.